openapi/components/schemas/Subscription/SubscriptionPause.yaml

```
@@ -0,0 +1,85 @@
+type: object
+required:
+  - subscriptionId
+properties:
+  id:
+    description: Pause identifier.
+    readOnly: true
+    allOf:
+      - $ref: ../ResourceId.yaml
+  subscriptionId:
+    description: Identifier of the paused subscription.
+    allOf:
+      - $ref: ../ResourceId.yaml
+  status:
+    description: The pause status.
+    type: string
+    readOnly: true
+    enum:
+      - pending
+      - ongoing
+      - revoked
+      - finished
+  pausedBy:
+    description: Who did the pause.
+    type: string
+    default: customer
+    enum:
+      - merchant
+      - customer
+  description:
+    description: Pause reason description in free form.
+    type: string
+    maxLength: 255
+    nullable: true
+  effectiveTime:
+    description: >
+      The date from which the service period will pause.
+
+      This time should be later than now, if it is earlier then current time is taken.
+
+      If omitted, it will default to the current time.
+    type: string
+    format: date-time
+    nullable: true
+  endTime:
+    description: >
+      The time when the pause ends and the subscription resumes its billing.
+
+      In order to resume a subscription now, use current time or past time (will default to now).
+      Empty endTime will result in indefinitely paused subscription.
+    type: string
+    format: date-time
+    nullable: true
+  timeRemaining:
+    description: >
+      Time in seconds between the pause end time and the renewal time.
+      By default, it's computed as the time between the pause and the renewal.
+
+      For instance, you have a $30/month subscription and 30-day month.
+      If you pause on day 20 of the subscription, there are 10 days unused.
+      Therefore, the subscription would renew in 10 days after resuming.
+      The upcoming invoice would not change the amount for invoiced period, it remains $30.
+      There may be a billing anchor for the subscription,
+      the renewal time after a pause shift should behave as if the renewal time moved forward with Order API
+      (the billing anchor is ignored for that period then).
+    type: number
```

```suggestion

      If the end time is null, this will be 0 and the renewal time is not changed.
    type: number
```

---

@2e3s, why should it be zero? We must know time remaining from pause start to original renewal time.
For 30-day month, if you pause at day 20 for unknown time, you still have 10 days remaining

---

Because time remaining is `renewalTime - endTime`.

---

Hence this is an improved suggestion:
```suggestion

      If the end time is null, this will also be null and the renewal time is not changed.
    type: number
    nullable: true
```

---

> Because time remaining is renewalTime - endTime.

I thought we use original renewal time here and not the time that is shifted